element.go

```diff
@@ -342,29 +342,43 @@ func readRawItem(d *dicomio.Decoder) ([]byte, bool) {
 	return d.ReadBytes(int(vl)), false
 }
 
+type NativeFrame struct {
+	Data          [][]int // Slice of pixels that each have multiple values
+	Rows          int
+	Cols          int
+	BitsPerSample int
+}
+
+type EncapsulatedFrame struct {
+	Data []byte
+}
+
+// Frame represents a single encapsulated or native image frame
+type Frame struct {
+	IsEncapsulated    bool
+	EncapsulatedFrame EncapsulatedFrame
```

Comment:
We can flatten the `Frame` data structure if we just dump `EncapsulatedFrames []byte` and `NativeFrames [][]int` inside the `Frame` and bring along other attributes like `Rows`, `Cols`, and `BitsPerSample` along for each `Frame`...even though those attributes are really only _needed_ for work with NativeFrames

Reply:
It turns out those attributes are required for both encapsulated and native images so including them in the top level of `Frame` would not be too bad. They just make less sense for encapsulated images to have included with the Frame itself, but that's okay...

Reply:
for now, will likely leave as is, if a refactor is needed it can happen later. Including those elements with every `Frame` doesn't really add value for `EncapsulatedFrame`s since that information already exists in the encapsulated frame data. We also assume that clients have the previously parsed elements if they happen to need those elements, so no need to duplicate storage of that data for all types of frames.